app/src/ui/pull-request-quick-view.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import { PullRequest } from '../models/pull-request'
+import { PullRequestBadge } from './branches'
+import { Dispatcher } from './dispatcher'
+import { Button } from './lib/button'
+import { SandboxedMarkdown } from './lib/sandboxed-markdown'
+import { Octicon } from './octicons'
+import * as OcticonSymbol from './octicons/octicons.generated'
+
+interface IPullRequestQuickViewProps {
+  readonly dispatcher: Dispatcher
+  readonly pullRequest: PullRequest
+
+  /** When mouse leaves the PR quick view */
+  readonly onMouseLeave: () => void
+}
+
+export class PullRequestQuickView extends React.Component<
+  IPullRequestQuickViewProps,
+  {}
+> {
+  private baseHref = 'https://github.com/'
+
+  private renderHeader = (): JSX.Element => {
```

Comment:
Do we need to specify a return type here or can we rely on inference?

```suggestion
  private renderHeader = () => {
```

Reply:
Curious, I think you are the first person I have met to ask me to remove a return type. :D Do you just find that it clutters things/ do you think it just isn't useful for things like render which are not typically reusable methods?

Reply:
Hah, yeah, I can see how it runs a bit counter to my strong preference for static typing in general but we don't gain anything by including it here and explicitly typing the return value as JSX.Element could even cause us to have to change it down the line should we want to return a string for example (or make it optional, i.e. able to return a falsy value).

I think specifying explicit return types is great when we either want to codify an external requirement or when we want to make absolutely certain that we always return an element (though this is a bigger issue for codebases which allow implicit returns).

When I see explicit return types like this it makes me wonder what the special constraint is that caused us to add them and can make me concerned about changing the return type. All that said the primary reason I rarely write them myself is just because I'm lazy and because the compiler knows better what the return type is than I do.